### PR TITLE
Fix indicator warm-up window seed

### PR DESCRIPTION
**Pull request by aiSynergy37**
What: Clear indicator windows after reset during indicator history warm-up.

Why: `IndicatorHistory` and `WarmUpIndicator` could leave the reset default value in an expanded public indicator window before replaying historical samples.

How: Reset the indicator window immediately after `indicator.Reset()` in the history replay path so only real historical samples populate the public window.

Edge cases handled: Expanded windows no longer retain `DateTime.MinValue` seed points; the current indicator value remains at `Window[0]` after warm-up.

Tests added/updated: Added `AlgorithmIndicatorsTests.WarmUpIndicatorDoesNotLeaveDefaultValueInWindow`. Build validation completed for `Algorithm/QuantConnect.Algorithm.csproj` and `Indicators/QuantConnect.Indicators.csproj`; focused local test execution was blocked by Windows Application Control on the test assembly.

### CODE REVIEW

File: Algorithm/QCAlgorithm.Indicators.cs [hunk 1]
```diff
@@ -4455,6 +4455,7 @@ private IndicatorHistory IndicatorHistory<T>(IndicatorBase<T> indicator, IEnumer
         {
             // Reset the indicator
             indicator.Reset();
+            indicator.Window.Reset();
```
Comment (Martin-Molinero) [inline Algorithm/QCAlgorithm.Indicators.cs:4458]: `indicator.Reset()` is calling `indicator.Window.Reset()`
